test-resources/test_turn_lanes.json

```diff
@@ -1246,5 +1246,77 @@
     "expectedResults": {
       "603065820": "[MUTE] C:+C|TSLR"
     }
+  },
+  {
+    "testName": "73.1 Lane guidance has difficulty suggesting to go straight ahead https://github.com/osmandapp/OsmAnd/issues/18158",
+    "startPoint": {
+      "latitude": 45.534058,
+      "longitude": 35.54116
+    },
+    "endPoint": {
+      "latitude": 45.53585,
+      "longitude": 35.538643
+    },
+    "expectedResults": {
+      "9896643": "TR:TL|C|+TR,C"
+    }
+  },
+  {
+    "testName": "73.2 Lane guidance has difficulty suggesting to go straight ahead https://github.com/osmandapp/OsmAnd/issues/18158",
+    "startPoint": {
+      "latitude": 45.534058,
+      "longitude": 35.54116
+    },
+    "endPoint": {
+      "latitude": 45.534973,
+      "longitude": 35.53942
+    },
+    "expectedResults": {
+      "788401717": "TR:TL|C|+TR,C"
+    }
+  },
+  {
+    "testName": "73.3 Lane guidance has difficulty suggesting to go straight ahead https://github.com/osmandapp/OsmAnd/issues/18158",
+    "ignore":true,
+    "startPoint": {
+      "latitude": 45.534058,
+      "longitude": 35.54116
+    },
+    "endPoint": {
+      "latitude": 45.53525,
+      "longitude": 35.537155
+    },
+    "expectedResults": {
+      "711709168": "[MUTE] C:TL|+C|+C,TR"
+    }
+  },
+  {
+    "testName": "74. Invalid right turn direction with lanes https://github.com/osmandapp/OsmAnd/issues/18853",
+    "ignore":true,
+    "startPoint": {
+      "latitude": 45.535507,
+      "longitude": 35.568783
+    },
+    "endPoint": {
+      "latitude": 45.534256,
+      "longitude": 35.568367
+    },
+    "expectedResults": {
+      "761885309": "[MUTE] C:+C,TSLL|TR"
+    }
+  },
+  {
+    "testName": "75. Invalid right turn direction with lane https://github.com/osmandapp/OsmAnd/issues/19041",
```

Comment:
2 test variations - https://www.openstreetmap.org/way/588754800 only [C] and [C|TR]